### PR TITLE
Add function to allow looking up app based on name and owner username

### PR DESCRIPTION
**Pull request by CameronDiver**
This can be useful for looking up shared applications when the user in
question has an application of the same name as the one that is shared.

Connects-to: #349
Change-type: minor
Signed-off-by: Cameron Diver <cameron@resin.io>
---- Autogenerated Waffleboard Connection: Connects to #349

### CODE REVIEW

File: lib/models/application.coffee [hunk 1]
```diff
@@ -149,6 +149,53 @@ getApplicationModel = (deps, opts) ->
...
+	#
+	# @example
+	# resin.models.application.getAppWithOwner('MyApp', 'MyUser').then(function(application) {
+	#		console.log(application);
+	# });
+	###
+	exports.getAppWithOwner = (appName, owner, options = {}, callback) ->
+		callback = findCallback(arguments)
+
+		a.pine.get
```
Comment (pimterry) [inline lib/models/application.coffee]: What is `a`? I'm surprised this code works...

File: lib/models/application.coffee [hunk 2]
```diff
@@ -149,6 +149,53 @@ getApplicationModel = (deps, opts) ->
...
+						$eq: [
+							$tolower: $: 'app_name'
+							appName
+						]
+					expand:
+						user:
+							$filter:
+								$eq: [
+									$tolower: $: 'username'
+									owner
```
Comment (pimterry) [inline lib/models/application.coffee]: We should `toLowerCase` both the owner and appName here, if we're going to compare them to `$tolower`.

File: lib/models/application.coffee (hunk 1 above)
Comment (CameronDiver) [inline lib/models/application.coffee]: Oh shoot. I've copied the debugging code out of another file. It's essentially the same, but with different vars... Sorry for the noise.

File: lib/models/application.coffee (hunk 2 above)
Comment (CameronDiver) [inline lib/models/application.coffee]: Same as above, I've copied the wrong code...

**Comment by CameronDiver** [comment]
@pimterry I was using resin-sdk-preconfigured to do the testing and accidentally committed the wrong code! Sorry about that. Anyway I've added the correct code and added some tests for the new method.

File: lib/models/application.coffee [hunk 3]
```diff
@@ -149,6 +149,56 @@ getApplicationModel = (deps, opts) ->
...
+	#
+	# @param {String} appName - application name
+	# @param {String} owner - The owner's username
+	# @param {Object} [options={}] - extra pine options to use
+	# @fulfil {Object} - application
+	# @returns {Promise}
+	#
+	# @example
+	# resin.models.application.getAppWithOwner('MyApp', 'MyUser').then(function(application) {
+	#		console.log(application);
```
Comment (emirotin) [inline lib/models/application.coffee]: Unneeded double tab?

File: tests/integration/models/application.spec.coffee [hunk 4]
```diff
@@ -15,6 +15,13 @@ describe 'Application Model', ->
 				promise = resin.models.application.getAll()
 				m.chai.expect(promise).to.become([])
 
+		describe 'resin.models.application.getAppWithOwner()', ->
+
+			it 'should eventually reject', ->
+				promise = resin.models.application.getAppWithOwner('testapp', 'FooBar')
+				m.chai.expect(promise).to.be.rejected
+
+
```
Comment (emirotin) [inline tests/integration/models/application.spec.coffee]: Extra empty line

**Pull request merged**